xbmc/pvr/PVRContextMenus.cpp

```diff
@@ -0,0 +1,785 @@
+        return true;
+
+      return false;
+    }
+
+    bool PlayChannel::Execute(const CFileItemPtr &item) const
+    {
+      return CPVRGUIActions::GetInstance().SwitchToChannel(
+        item, CSettings::GetInstance().GetBool(CSettings::SETTING_PVRPLAYBACK_PLAYMINIMIZED), false /* bCheckResume */);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Resume play recording
+
+    std::string ResumePlayRecording::GetLabel(const CFileItem &item) const
+    {
+      return CPVRGUIActions::GetInstance().GetResumeLabel(item);
+    }
+
+    bool ResumePlayRecording::IsVisible(const CFileItem &item) const
+    {
+      CPVRRecordingPtr recording;
+
+      const CEpgInfoTagPtr epg(item.GetEPGInfoTag());
+      if (epg)
+        recording = epg->Recording();
+
+      if (!recording)
+        recording = item.GetPVRRecordingInfoTag();
+
+      if (recording)
+        return !recording->IsDeleted() && !CPVRGUIActions::GetInstance().GetResumeLabel(item).empty();
+
+      return false;
+    }
+
+    bool ResumePlayRecording::Execute(const CFileItemPtr &item) const
+    {
+      item->m_lStartOffset = STARTOFFSET_RESUME; // must always be set if PlayRecording is called with bCheckResume == false
+      return CPVRGUIActions::GetInstance().PlayRecording(item, false /* bPlayMinimized */, false /* bCheckResume */);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Play recording
+
+    std::string PlayRecording::GetLabel(const CFileItem &item) const
+    {
+      const CPVRRecordingPtr recording(item.GetPVRRecordingInfoTag());
+      if (recording)
+      {
+        if (!recording->IsDeleted())
+        {
+          if (CPVRGUIActions::GetInstance().GetResumeLabel(item).empty())
+            return g_localizeStrings.Get(208); /* Play */
+          else
+            return g_localizeStrings.Get(12021); /* Play from beginning */
+        }
+      }
+
+      return g_localizeStrings.Get(19687); /* Play recording */
+    }
+
+    bool PlayRecording::IsVisible(const CFileItem &item) const
+    {
+      CPVRRecordingPtr recording;
+
+      const CEpgInfoTagPtr epg(item.GetEPGInfoTag());
+      if (epg)
+        recording = epg->Recording();
+
+      if (!recording)
+        recording = item.GetPVRRecordingInfoTag();
+
+      if (recording)
+        return !recording->IsDeleted();
+
+      return false;
+    }
+
+    bool PlayRecording::Execute(const CFileItemPtr &item) const
+    {
+      item->m_lStartOffset = 0; // must always be set if PlayRecording is called with bCheckResume == false
+      return CPVRGUIActions::GetInstance().PlayRecording(item, false /* bPlayMinimized */, false /* bCheckResume */);
+    }
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Mark watched
+
+    std::string MarkWatched::GetLabel(const CFileItem &item) const
+    {
+      return g_localizeStrings.Get(16103); /* Mark as watched */
```

Comment:
Cool. I will change the code to use <code>CStaticContextMenuAction</code> where possible.